### ISSUE TITLE
Add support for JSON:API includes in the Drupal source plugin

### ISSUE THREAD

**Issue by klaasvw** [gatsbyjs/gatsby#21379]
## Summary

Drupal's JSON:API module supports [includes](https://www.drupal.org/docs/8/modules/jsonapi/includes), a method for including relationships data directly, without requiring the consumer to follow the relationship links. 

Supporting includes can be very powerful for big Drupal sites, especially those with complex relationships or using modules like Paragraphs. When combined with `disallowedLinkTypes` it enables you to fetch only the data your Gatsby site needs at build time.

### Basic example

Take for example a Drupal website with the following architecture:

- Two node types `page` and `news`
- Two paragraph types `text` and `image`
- An entity reference field `field_content` allowing unlimited `text` and `image` paragraphs, used by both the `page` and `news` content types.

Our Gatsby site only needs to display `news`, so we use the following config for the `gatsby-source-drupal` pugin:

```javascript
module.exports = {
  plugins: [
    {
      resolve: `gatsby-source-drupal`,
      options: {
        baseUrl: `https://live-contentacms.pantheonsite.io/`,
        // skip the node--page resource type.
        disallowedLinkTypes: [`self`, `describedby`, `node--page`],
      },
    },
  ],
}
```

The result is that at build time Gatsby will get data for:

- All `news` nodes
- All `text` and `image` paragraphs

That means, also all the `text` and `image` paragraphs used by `page` nodes, even though we're not fetching them.

If we support JSON:API includes we can do this instead:

```javascript
module.exports = {
  plugins: [
    {
      resolve: `gatsby-source-drupal`,
      options: {
        baseUrl: `https://live-contentacms.pantheonsite.io/`,
        // skip the node--page resource type.
        disallowedLinkTypes: [`self`, `describedby`, `node--page`, `paragraph--text`, `paragraph--image`],
        filters: {
          "node--news": "include=field_content",
        },
      },
    },
  ],
}
```
At build time this will fetch:

- All `news` nodes
- Only the paragraphs includes in the `news` nodes

### Motivation

This is especially interesting for Drupal sites with many relationships between entities and Gatsby sites that only want to consume a smaller subset of these entities.

In one example we cut down build time from **35 minutes to 3 minutes** on a Drupal website using many paragraph types combined with many node types where the Gatsby site only consumed a single node type.

**Comment by kmajzlik** [comment]
Absolutelly looking for this.
Node -> Paragraphs -> Media -> Files.
I want to scan only node resource (and load only needed paragraphs/media/files).
Will test PR later this week.

**Comment by kmajzlik** [comment]
And just small note: look at jsonapi_extras module and its submodule jsonapi_defaults - then you dont need to specify it in Gatsby.

**Comment by smthomas** [comment]
This was published in gatsby-source-drupal@3.3.19. Refer to #21380 and updated documentation at https://www.npmjs.com/package/gatsby-source-drupal#disallowed-link-types
